README.md

```diff
@@ -32,21 +34,73 @@ The API is available on the `cordova.plugins.idfa` global object.
 
-- `limitAdTracking`: `boolean` - Whether usage of advertising id is allowed by user.
+- `isTrackingLimited`: `boolean` - Whether usage of advertising id is allowed by user.
 - `idfa`: `string` (_iOS only_) - Identifier for advertisers.
-- `trackingTransparencyStatus` (_iOS only_): `"NotAvailable"` | `"Authorized"` | `"Denied"` | `"Restricted"` | `"NotDetermined"` -
-   Tracking transparency status, available on iOS 14+ devices. On devices with iOS < 14 the value will always be
-   `"NotAvailable"`. For the meaning of other values see [the tracking transparency API docs](https://developer.apple.com/documentation/apptrackingtransparency/attrackingmanagerauthorizationstatus).
+- `trackingPermission` (_iOS only_): [`TrackingPermission`](#trackingpermission)
+   Tracking permission status, available on iOS 14+ devices. On devices with iOS < 14 the value will
+   always be `null`.
 - `aaid`: `string` (_Android only_) - Android advertising ID.
 
+### requestPermission()
+
+_(iOS only)_ A one-time request to authorize or deny access to app-related data that can be used for
+tracking the user or the device. See [Apple's API docs][requesttrackingauthorization-api-url]
+for more info on the dialog presented to the user. Available only for iOS 14+ devices.
+
+Returns a `Promise<`[`TrackingPermission`](#trackingpermission)`>`. On devices
+with iOS < 14 the promise will always resolve with `null`.
+
+**Note:** You should make sure to set the
+[`NSUserTrackingUsageDescription`][nsusertrackingusagedescription-api-url] key in your app's
+Information Property List file, otherwise your app will crash when you use this API.
+You can do it with the following code in your Cordova project's `config.xml`:
+```xml
+<platform name="ios">
+    <edit-config target="NSUserTrackingUsageDescription" file="*-Info.plist" mode="merge">
+        <string>My tracking usage description</string>
+    </edit-config>
+</platform>
+```
+
+### TrackingPermission
+
+An `object` containing all possible tracking permission values returned by [`getInfo()#trackingPermission`](#getinfo)
+and [`requestPermission()`](#requestPermission).
+
+For the meaning of the values see [the tracking transparency API docs][authorizationstatus-api-url]:
+
+- `Authorized`: `'Authorized'`
+- `Denied`: `'Denied'`
+- `Restricted`: `'Restricted'`
+- `NotDetermined`: `'NotDetermined'`
+
 ## Example
 
 ```js
-cordova.plugins.idfa.getInfo().then(function(info) {
-    if (!info.limitAdTracking) {
-        console.log(info.idfa || info.aaid);
-    }
-});
+const idfaPlugin = cordova.plugins.idfa;
+const TrackingPermission = idfaPlugin.TrackingPermission;
+
+idfaPlugin.getInfo()
+    .then(info => {
+        if (!info.isTrackingLimited) {
+            return info.idfa || info.aaid;
+        } else if (info.trackingPermission === TrackingPermission.NotDetermined) {
+            return idfaPlugin.requestPermission().then(result => {
+                if (result !== TrackingPermission.Authorized) {
```

Comment:
Can be simplified:
```js
if (result === TrackingPermission.Authorized) {
    return idfaPlugin.getInfo().then(info => {
                    return info.idfa || info.aaid;
                });
}
```

Reply:
Well, this was on purpose. This way we short-circuit invalid state handling and leave the valid state handling outside the if-block. Still want me to change this?

Reply:
> leave the valid state handling outside the if-block

I think valid state handling should be in the latest `then` block (line 101).

Reply:
I don't really understand what you mean. We can't move the check in line 91 to line 102, the call to `idfaPlugin.getInfo()` depends on this check. There is still a notion of validity there, it is basically invalid/unnecessary to call `idfaPlugin.getInfo()` if `result !== idfaPlugin.TRACKING_PERMISSION_AUTHORIZED`.

Could you please elaborate on this?

Reply:
Sorry I misunderstood. I was thinking not about valid handler, but success handler:
```js
cordova.plugins.idfa.getInfo(info => {
    if (!info.trackingLimited) {
        return info.idfa || info.aaid;
    } else if (info.trackingPermission === "NotDetermined") {
        return cordova.plugins.idfa.requestPermission(result => {
            if (result === TrackingPermission.Authorized) {
                return cordova.plugins.idfa.getInfo(info => {
                    return info.idfa || info.aaid;
                });
            }
        });
    }
}).then(idfaOrAaid => {
    if (idfaOrAaid) {
        // success handler
    }
});
```

So we don't really need "to leave the valid state handling outside the if-block." because the success handler is later anyway.